hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -129,6 +129,9 @@ public static ClientMessage create() {
 
     public static ClientMessage createForEncode(int initialCapacity) {
         initialCapacity = QuickMath.nextPowerOfTwo(initialCapacity);
+        if (initialCapacity < 0) {
+            initialCapacity = Integer.MAX_VALUE;
+        }
         if (USE_UNSAFE) {
```

Comment:
would it not be better to have this fix in QuickMath.nextPowerOfTwo? I can see it's called from some other places also